File: single-language-analysis/dataset-files/python/test_smell_048.py
import pytest

class TestOOoBean:
    def test_smell_048(self):
        f = None
        try:
            f = WriterFrame(100, 100, 500, 400, False, connection.get_component_context())
            f.go_to_start()
            f.page_down()
            time.sleep(1)

            capturer = ScreenComparer(100, 100, 500, 400)
            capturer.grab_one()
            for i in range(100):
                f.remove_o_o_bean()
                f.add_o_o_bean()

            f.go_to_start()
            f.page_down()
            time.sleep(get_sleep_time(200))
            capturer.grab_two()

            if not capturer.compare():
                pytest.fail("Painting error: adding and removing OOoBean repeatedly to java.lang.Frame.")
                capturer.write_images()

            if not f.check_uno_frame_position():
                pytest.fail("Sizing error.")

        finally:
            if f is not None:
                f.dispose()
            if not is_windows():
                time.sleep(10)

File: single-language-analysis/dataset-files/java/TestSmell048.java
// Original URL: https://github.com/alibaba/alios-libreoffice/blob/e9a6cc08a12b5133a9dccf337d16f9a534f57f1e/bean/qa/complex/bean/OOoBeanTest.java#L324-L371

public class TestSmell048 {
    @Test public void test5() throws Exception
    {
        WriterFrame f = null;
        try
        {
            f = new WriterFrame(100, 100, 500, 400, false, connection.getComponentContext());
            f.goToStart();
            f.pageDown();
            Thread.sleep(1000);

            ScreenComparer capturer = new ScreenComparer(100,100,500,400);
            capturer.grabOne();
            for (int i = 0; i < 100; i++)
            {
                f.removeOOoBean();
                f.addOOoBean();
            }

            f.goToStart();
            f.pageDown();
            Thread.sleep(getSleepTime(200));
            capturer.grabTwo();

            if (!capturer.compare())
            {
                fail("Painting error: adding and removing OOoBean " +
                       "repeatedly to java.lang.Frame.");
                capturer.writeImages();
            }

            if (!f.checkUnoFramePosition())
            {
                fail("Sizing error.");
            }

        }
        finally
        {
            if (f != null)
            {
                f.dispose();
            }
            if (!isWindows())
            {
                Thread.sleep(10000);
            }
        }
    }
}
